### ISSUE TITLE
Query string enum argument not using schema reference

### ISSUE THREAD

**Issue by youfoundkris** [domaindrivendev/Swashbuckle.AspNetCore#1160]
## VERSION: ##
Swashbuckle.AspNetCore 4.0.1

## STEPS TO REPRODUCE: ##

1. Create two controller methods that use the same enum as a query string parameter.
2. Create another controller method that returns a model that contains a property with the same enum.
3. Configure DescribeAllEnumsAsStrings
4. Configure UseReferencedDefinitionsForEnums

## EXPECTED RESULT: ##
The generated documentation uses schema references for the for all three usages of the enum.

## ACTUAL RESULT: ##
The generated documentation uses an inline enum for the query string arguments.

```
          "parameters": [
            {
              "type": "string",
              "name": "myenum",
              "in": "query",
              "description": "my desc",
              "enum": [
                "NotSet",
                "ValueA",
                "ValueB",
                "ValueC"
              ]
            }
```

## ADDITIONAL DETAILS ##
When using NSwag to generate a client, this results in multiple enums being generated:

- MyEnum
- MyEnum2
- MyEnum3

**Comment by domaindrivendev** [comment]
This is a limitation to the Swagger/OAI 2x specification. As you can see from the JSON above, parameters don't have schemas. Instead, they have their own (similar) descriptive properties like `type` and `enum`. These are implicitly part of the `parameter` and therefore can't be delegated to a referenced dschema.

**Comment by youfoundkris** [comment]
Thanks for clearing this up!
How is this addressed in Swashbuckle 5.0.0?

**Comment by domaindrivendev** [comment]
The latest pre-release version generates enums as referenced schemas out-of-the-box
